### ISSUE TITLE
is datastore support 16KB page size？

### ISSUE THREAD

**Issue by peienliudao** [googlecodelabs/android-datastore#76]
datastore contains libdatastore_shared_counter.so,
this so file support 16kb page size in android 15?

**Comment by peienliudao** [comment]
androidx.datastore:datastore-preferences:1.1.1

**Comment by alexander-m-ilyin** [comment]
The 1.1.7 and 1.2.0 still have no 16k support. The 1.1.1 looks ok, so I think there is some regression in the latest releases.

**Comment by anonym24** [comment]
The annoying part is that yesterday I was able to upload updates on Google Play for other apps with the same version (v1.2.0), so it seems I will have to make new updates for them again. 
Basically, did they stop accepting this version today or what? Now I can't upload it for another app.

Google Play now doesn't accept new updates with datastore v1.2.0 because of 16K issue.

Analyze APK from Android Studio of the latest version "Panda 2 | 2025.3.2 Build #AI-253.30387.90.2532.14935130, built on February 25, 2026" reports:

`libdatastore_shared_counter.so - RELRO is not a suffix and its end is not 16 KB aligned`

<img width="1080" height="528" alt="Image" src="https://github.com/user-attachments/assets/65ced636-8245-4a16-bbcf-dad4cc6bad7d" />

**Comment by alexander-m-ilyin** [comment]
> The annoying part is that yesterday I was able to upload updates on Google Play for other apps with the same version (v1.2.0), so it seems I will have to make new updates for them again. Basically, did they stop accepting this version today or what? Now I can't upload it for another app.
> 
> Google Play now doesn't accept new updates with datastore v1.2.0 because of 16K issue.
> 

Yes, we faced the same issue with 1.1.7 and 1.2.0, so downgraded to 1.1.1.

**Comment by andrespelaezp** [comment]
For anyone still looking at this issue, version 1.2.1 now fixed [this issue](https://android.googlesource.com/platform/frameworks/support/+/00bc339d026c39791f891ea9db2313c1fcdc97f9). List of commits: https://android.googlesource.com/platform/frameworks/support/+log/9c556fd714d249b153cbe3ddce61e9eb43f77ffc..03aea68c431abd3fa436e9f8fa9b9cda12f18334
